boa/deployments.py

```diff
@@ -123,15 +129,19 @@ def insert_deployment(self, deployment: Deployment):
 
     def _get_deployments_from_sql(self, sql_query: str, parameters=(), /):
         cur = self.db.execute(sql_query, parameters)
-        ret = [Deployment.from_sql_tuple(item) for item in cur.fetchall()]
-        return ret
+        return (Deployment.from_sql_tuple(item) for item in cur)
 
     def _get_fieldnames_str(self) -> str:
         return ",".join(field.name for field in fields(Deployment))
 
-    def get_deployments(self) -> list[Deployment]:
+    def get_deployments_iter(self) -> Generator[Deployment]:
```

Comment:
given the builtin is `iter` that's used as a verb usually
```suggestion
    def iter_deployments(self) -> Generator[Deployment]:
```

Reply:
good point. i've removed the `get_deployments_iter` function entirely and changed `get_deployments` to return the iterator instead of a list. https://github.com/vyperlang/titanoboa/pull/324/commits/ffdf36f974e10cdfd3336c6d63ca7136b4dc2560